### PR TITLE
Refactored BMS driver for boolean support

### LINKED ISSUE #302: bms driver: get bool status fields
publish as Float64 and the cast to bool in power_state_aggregator.py

```python
charging= (int(msg.data) & (1<<BIT_NR)) == (1<<BIT_NR)
```

### PR DESCRIPTION
**Pull request by mathias-luedtke**
fixes #302 

Needs to be tested on hardware.

review with https://github.com/ipa320/cob_driver/pull/303/files?w=1

### CODE REVIEW

**Comment by mathias-luedtke** [comment]
fixed some bugs during hardware testing

**Review by benmaidel — APPROVED** [review]

File: cob_bms_driver/include/cob_bms_driver/cob_bms_driver_node.h [hunk 1]
```diff
@@ -1,101 +1,104 @@
...
+
+    //function that polls BMS (bms_id_to_poll_ is used here!).
+    //Also, this function sleeps for time given by poll_period_for_two_ids_in_ms_ to ensure BMS is polled at the desired frequency
+    void pollBmsForIds(const uint16_t first_id, const uint16_t second_id);
+
+    //callback function to handle all types of frames received from BMS
+    void handleFrames(const can::Frame &f);
+
+            //updates the diagnostics data with the new data received from BMS
+            void produceDiagnostics(diagnostic_updater::DiagnosticStatusWrapper &stat);
```
Comment (floweisshardt) [inline cob_bms_driver/include/cob_bms_driver/cob_bms_driver_node.h]: tabs vs spaces

**Review by floweisshardt — CHANGES_REQUESTED** [review]

File: cob_bms_driver/include/cob_bms_driver/cob_bms_driver_node.h (hunk 1 above)
Comment (mathias-luedtke) [reply cob_bms_driver/include/cob_bms_driver/cob_bms_driver_node.h]: not tabs, just indention is wrong.
I am not sure when I can fix the white space issues.

**Review by NadiaHG — APPROVED** [review]
Test on cob4-7: OK

**Review by floweisshardt — APPROVED** [review]

**Pull request merged**